pkg/client/restclient/request.go

```diff
@@ -120,20 +120,19 @@ type Request struct {
 }
 
 // NewRequest creates a new request helper object for accessing runtime.Objects on a server.
-func NewRequest(client HTTPClient, verb string, baseURL *url.URL, versionedAPIPath string, content ContentConfig, serializers Serializers, backoff BackoffManager, throttle flowcontrol.RateLimiter) *Request {
+func NewRequest(client HTTPClient, verb string, urlProvider URLProvider, versionedAPIPath string, content ContentConfig, serializers Serializers, backoff BackoffManager, throttle flowcontrol.RateLimiter) *Request {
 	if backoff == nil {
 		glog.V(2).Infof("Not implementing request backoff strategy.")
 		backoff = &NoBackoff{}
 	}
-
 	pathPrefix := "/"
-	if baseURL != nil {
-		pathPrefix = path.Join(pathPrefix, baseURL.Path)
+	if u := urlProvider.Get(); u != nil {
```

Comment:
Ah - does this mean this is in fact _not_ a URLProvider, but a HostProvider?  If we can't mix Path values, we probably shouldn't be allowing it in the contract.

I think the answer is likely to push pathPrefix down so we resolve it whenever we call urlprovider.Get